include/envoy/event/dispatcher.h

```diff
@@ -149,8 +149,9 @@ class Dispatcher {
   /**
    * Allocate a timer. @see Timer for docs on how to use the timer.
    * @param cb supplies the callback to invoke when the timer fires.
+   * @param cb supplies an optional scope for the duration of the alarm.
    */
-  virtual Event::TimerPtr createTimer(TimerCb cb) PURE;
+  virtual Event::TimerPtr createTimer(TimerCb cb, const ScopeTrackedObject* object = nullptr) PURE;
```

Comment:
High level question: Do you think we should pass the scope as part of timer creation or rather when we enable the timer? My concern is I wonder if we might at some point have a different scope for different deadlines on the same timer? TBH this seems unlikely, but at the same time I worry about having to redo this interface if/when that happens? WDYT?

/wait-any

Reply:
Should that go for passing the dispatcher through as well, or should that stay in the constructor?

Reply:
Sorry what do you mean?

Reply:
Currently in the constructor we're passing through the dispatcher (also needed for scope setting) and the scope.
Preference for moving both to enableTimer, or leaving dispatcher as a one-time thing?

Reply:
IMO the dispatcher should not change so constructor is fine. WDYT?

Reply:
doesn't change today, and if we ever have alarms that switch threads I think the constructor is scoped enough we can do the work then.